src/Components/Facility/ConsultationDetails/index.tsx

```diff
@@ -149,6 +147,14 @@ export const ConsultationDetails = (props: any) => {
                 : "No",
               is_vaccinated: patientData.is_vaccinated ? "Yes" : "No",
             };
+            if (
+              facilityId != data.facility_object?.id ||
+              patientId != data.id
```

Comment:
`data` here represents "Patient" object. Patient's facility need not always be the same as "Consultation" object's facility, as a patient can have multiple consultations, in different facilities. 

Hence this won't work as expected for an older consultation of a patient if the previous consultation belongs to a different facility than the current active consultation's facility right?

Do the check against `consultation.facility` instead of `patient (data).id`